Mismatched type error in private.rs

I am unable to build godot-rust from the master branch due to a mismatched type error in private.rs.

Error with context:
```
   Compiling gdnative-core v0.8.1 (C:\projects\godot-rust\gdnative-core)
error[E0308]: mismatched types
  --> gdnative-core\src\private.rs:56:21
   |
56 |         if type_ == crate::sys::GDNATIVE_API_TYPES_GDNATIVE_CORE
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `i32`
   |
help: you can convert an `i32` to `u32` and panic if the converted value wouldn't fit
   |
56 |         if type_ == crate::sys::GDNATIVE_API_TYPES_GDNATIVE_CORE.try_into().unwrap()
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: could not compile `gdnative-core`.
```
Build tested and failed with commits 3e69d97df5244de49271f4097f2cee163db21dcc (where the new Godot version-checking code was added) and 5e6bac40fd72a9abb62c805f133bb31ef730f889 (where the code was merged to master).
Build tested and succeeded with commit 208088e518750e0d37347014ca4f2ad7ac8bece7 (immediately prior to the merge).

Additional information:
```
Default host: x86_64-pc-windows-msvc

stable-x86_64-pc-windows-msvc (default)
rustc 1.44.1 (c7087fe00 2020-06-17)
```

The CI build did succeed on 64-bit Windows 10, but I and another person were able to replicate this error on 64-bit Windows 10.

--- a/.travis.yml
+++ b/.travis.yml
@@ -75,22 +75,30 @@ matrix:
     - name: cargo test (stable, windows)
       os: windows
       rust: stable
-      env: CI_STAGE_CARGO_TEST=yes
+      env: CI_STAGE_CARGO_TEST=yes CI_MSVC_TEST=yes
+      before_script:
+        - rustup target add x86_64-pc-windows-msvc
 
     - name: cargo build --release (stable, windows)
       os: windows
       rust: stable
-      env: CI_STAGE_BUILD_RELEASE=yes
+      env: CI_STAGE_BUILD_RELEASE=yes CI_MSVC_TEST=yes
+      before_script:
+        - rustup target add x86_64-pc-windows-msvc
 
     - name: cargo test (nightly, windows)
       os: windows
       rust: nightly
-      env: CI_STAGE_CARGO_TEST=yes
+      env: CI_STAGE_CARGO_TEST=yes CI_MSVC_TEST=yes
+      before_script:
+        - rustup target add x86_64-pc-windows-msvc
 
     - name: cargo build --release (nightly, windows)
       os: windows
       rust: nightly
-      env: CI_STAGE_BUILD_RELEASE=yes
+      env: CI_STAGE_BUILD_RELEASE=yes CI_MSVC_TEST=yes
+      before_script:
+        - rustup target add x86_64-pc-windows-msvc
 
     - name: cargo test ios-simulator
       os: osx
@@ -127,10 +135,16 @@ script:
 
   - if [[ "$CI_STAGE_CARGO_TEST" == "yes" ]]; then
       cargo test --all --all-features;
+      if [[ "$CI_MSVC_TEST" == "yes" ]]; then
+        cargo test --target x86_64-pc-windows-msvc --all --all-features;
+      fi
     fi
 
   - if [[ "$CI_STAGE_BUILD_RELEASE" == "yes" ]]; then
       cargo build --release;
+      if [[ "$CI_MSVC_TEST" == "yes" ]]; then
+        cargo build --target x86_64-pc-windows-msvc --release;
+      fi
     fi
 
   - if [[ "$CI_STAGE_GODOT_TEST" == "yes" ]]; then

--- a/gdnative-core/src/private.rs
+++ b/gdnative-core/src/private.rs
@@ -53,7 +53,7 @@ unsafe fn check_api_compatibility(
         } = *api;
 
         // Godot 4 is not yet supported
-        if type_ == crate::sys::GDNATIVE_API_TYPES_GDNATIVE_CORE
+        if type_ as crate::sys::GDNATIVE_API_TYPES == crate::sys::GDNATIVE_API_TYPES_GDNATIVE_CORE
             && version.major == 1
             && version.minor == 3
         {

